File: GSO_Library/Models/Ensemble.cs
namespace GSO_Library.Models;

public class Ensemble
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Website { get; set; }
    public string? ContactInfo { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string? CreatedBy { get; set; }

    // Navigation properties
    public virtual ICollection<Performance> Performances { get; set; } = [];
}

File: GSO_Library.Web/src/types/ensemble.ts
import type { Performance } from './performance';

export interface Ensemble {
  id: number;
  name: string;
  description?: string;
  website?: string;
  contactInfo?: string;
  performances?: Performance[];
  createdAt: string;
  updatedAt: string;
  createdBy?: string;
}
